File: gcd/euclidean/src/EuclideanDemo.cpp
// C++
// 最大公約数: ユークリッドの互除法 (Euclidean)

#include <iostream>
#include <utility>

int gcd(int a, int b) {
    // bが0になるまで繰り返し計算する
    while (b) {
        std::tie(a, b) = std::make_pair(b, a % b);
    }
    return a;
}

int main() {
    std::cout << "Euclidean TEST -----> start" << std::endl;
    
    std::pair<int, int> input = {48, 18};
    std::cout << "\n  入力値: (" << input.first << ", " << input.second << ")" << std::endl;
    
    int output = gcd(input.first, input.second);
    std::cout << "\n  出力値: " << output << std::endl;
    
    std::cout << "\nEuclidean TEST <----- end" << std::endl;
    
    return 0;
}

File: gcd/euclidean/src/EuclideanDemo.kt
// Kotlin
// 最大公約数: ユークリッドの互除法 (Euclidean)

fun gcd(a: Int, b: Int): Int {
    // bが0になるまで繰り返し計算する
    var x = a
    var y = b
    while (y != 0) {
        val temp = y
        y = x % y
        x = temp
    }
    return x
}

fun main() {
    println("Euclidean TEST -----> start")
    val input = Pair(48, 18)
    println("\n  入力値: $input")
    val output = gcd(input.first, input.second)
    println("\n  出力値: $output")
    println("\nEuclidean TEST <----- end")
}